### PR TITLE
Update Portuguese Translation

### PR DESCRIPTION
**Pull request by FabioRosado**
# Description

Opsdroid has evolved a lot since I added the Portuguese translation. This PR aims to update all the logging messages to Portuguese.

I had to fix some compile errors being thrown when trying to update the catalog - basically we were using `%d:%s` and the translated message was using the old `%s, %s` so an error was being thrown on all the languages.

I've also noticed that the dialogflow was using string format on a log so I've changed that to use the proper format.

Fixes #<issue>

## Status
**READY** | ~~**UNDER DEVELOPMENT**~~ | ~~**ON HOLD**~~

## Type of change

_Please delete options that are not relevant._

- Documentation (fix or adds documentation)

# How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration._

- Tox - still green

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/locale/no/LC_MESSAGES/opsdroid.po [hunk 1]
```diff
@@ -18,143 +18,139 @@ msgstr ""
...
-#: opsdroid/core.py:408
+#: opsdroid/core.py:453
 msgid "Checking wit.ai..."
 msgstr "Sjekker wit.ai..."
 
-#: opsdroid/core.py:415
+#: opsdroid/core.py:460
+#, fuzzy
+msgid "Checking IBM Watson..."
+msgstr "Sjekker wit.ai..."
```
Comment (jacobtomlinson) [inline opsdroid/locale/no/LC_MESSAGES/opsdroid.po]: Should this be 
```suggestion
msgstr "Sjekker IBM Watson..."
```

File: opsdroid/locale/de/LC_MESSAGES/opsdroid.po [hunk 2]
```diff
@@ -18,143 +18,139 @@ msgstr ""
...
-#: opsdroid/core.py:408
+#: opsdroid/core.py:453
 msgid "Checking wit.ai..."
 msgstr "Prüfe wit.ai..."
 
-#: opsdroid/core.py:415
+#: opsdroid/core.py:460
+#, fuzzy
+msgid "Checking IBM Watson..."
+msgstr "Prüfe wit.ai..."
```
Comment (jacobtomlinson) [inline opsdroid/locale/de/LC_MESSAGES/opsdroid.po]:
```suggestion
msgstr "Prüfe IBM Watson..."
```

File: opsdroid/locale/de/LC_MESSAGES/opsdroid.po [hunk 3]
```diff
@@ -528,6 +509,16 @@ msgstr "Analysiere Eingabe: %s"
...
 
+#: opsdroid/connector/shell/__init__.py:19
+#, fuzzy
+msgid "Loaded shell connector"
+msgstr "Sqlite Datenbank connector geladen"
+
+#: opsdroid/connector/shell/__init__.py:112
+#, fuzzy, python-format
+msgid "Connecting to shell"
+msgstr "Mit sqlite %s verbunden"
```
Comment (jacobtomlinson) [inline opsdroid/locale/de/LC_MESSAGES/opsdroid.po]: Don't think this should mention sqlite?

File: opsdroid/locale/de/LC_MESSAGES/opsdroid.po [hunk 4]
```diff
@@ -592,7 +583,7 @@ msgstr ""
 
 #: opsdroid/connector/telegram/__init__.py:24
 #, fuzzy
-msgid "Loaded telegram connector"
+msgid "Loaded Telegram Connector"
 msgstr "Sqlite Datenbank connector geladen"
```
Comment (jacobtomlinson) [inline opsdroid/locale/de/LC_MESSAGES/opsdroid.po]: Why does this also mention sqlite?

File: opsdroid/locale/de/LC_MESSAGES/opsdroid.po [hunk 5]
```diff
@@ -614,8 +605,9 @@ msgid "Unable to delete webhook."
 msgstr ""
 
 #: opsdroid/connector/telegram/__init__.py:126
-msgid "Connecting to telegram"
-msgstr ""
+#, fuzzy, python-format
+msgid "Connecting to Telegram"
+msgstr "Mit sqlite %s verbunden"
```
Comment (jacobtomlinson) [inline opsdroid/locale/de/LC_MESSAGES/opsdroid.po]: Sqlite again?

File: opsdroid/locale/el/LC_MESSAGES/opsdroid.po [hunk 6]
```diff
@@ -7,334 +7,312 @@ msgid ""
...
-#: opsdroid/core.py:408
+#: opsdroid/core.py:453
 msgid "Checking wit.ai..."
 msgstr "Έλεγχος wit.ai..."
 
-#: opsdroid/core.py:415
+#: opsdroid/core.py:460
+#, fuzzy
+msgid "Checking IBM Watson..."
+msgstr "Έλεγχος wit.ai..."
```
Comment (jacobtomlinson) [inline opsdroid/locale/el/LC_MESSAGES/opsdroid.po]:
```suggestion
msgstr "Έλεγχος IBM Watson..."
```

File: opsdroid/locale/el/LC_MESSAGES/opsdroid.po [hunk 7]
```diff
@@ -538,6 +516,16 @@ msgstr "Απάντηση με: %s"
 msgid "Error - %s: Unable to respond"
 msgstr "Σφάλμα - %s: Αδυναμία ανταπόκρισης"
 
+#: opsdroid/connector/shell/__init__.py:19
+#, fuzzy
+msgid "Loaded shell connector"
+msgstr "Φορτώθηκε ο σύνδεσμος του telegram"
```
Comment (jacobtomlinson) [inline opsdroid/locale/el/LC_MESSAGES/opsdroid.po]: Should this say telegram?

File: opsdroid/locale/el/LC_MESSAGES/opsdroid.po [hunk 8]
```diff
@@ -538,6 +516,16 @@ msgstr "Απάντηση με: %s"
...
+
+#: opsdroid/connector/shell/__init__.py:112
+#, fuzzy
+msgid "Connecting to shell"
+msgstr "Σύνδεση στο Slack"
```
Comment (jacobtomlinson) [inline opsdroid/locale/el/LC_MESSAGES/opsdroid.po]: Should this say Slack?

File: opsdroid/locale/es/LC_MESSAGES/opsdroid.po [hunk 9]
```diff
@@ -18,143 +18,139 @@ msgstr ""
...
-#: opsdroid/core.py:408
+#: opsdroid/core.py:453
 msgid "Checking wit.ai..."
 msgstr "Comprobando wit.ai..."
 
-#: opsdroid/core.py:415
+#: opsdroid/core.py:460
+#, fuzzy
+msgid "Checking IBM Watson..."
+msgstr "Comprobando wit.ai..."
```
Comment (jacobtomlinson) [inline opsdroid/locale/es/LC_MESSAGES/opsdroid.po]:
```suggestion
msgstr "Comprobando IBM Watson..."
```

File: opsdroid/locale/no/LC_MESSAGES/opsdroid.po [hunk 10]
```diff
@@ -18,143 +18,139 @@ msgstr ""
...
 msgid "Checking luisai..."
 msgstr "Sjekker wit.ai..."
 
-#: opsdroid/core.py:401
-msgid "Checking Recast.AI..."
-msgstr "Sjekker Recast.AI..."
+#: opsdroid/core.py:446
+#, fuzzy
+msgid "Checking SAPCAI..."
+msgstr "Sjekker wit.ai..."
```
Comment (jacobtomlinson) [inline opsdroid/locale/no/LC_MESSAGES/opsdroid.po]:
```suggestion
msgstr "Sjekker SAPCAI..."
```

File: opsdroid/locale/no/LC_MESSAGES/opsdroid.po [hunk 11]
```diff
@@ -878,6 +874,29 @@ msgstr ""
 msgid "SAP Conversational AI score lower than min-score"
 msgstr ""
 
+#: opsdroid/parsers/watson.py:75
+#, fuzzy, python-format
+msgid "Watson response - %s"
+msgstr "Dialogflow respons - %s"
```
Comment (jacobtomlinson) [inline opsdroid/locale/no/LC_MESSAGES/opsdroid.po]: Watson?

**Comment by FabioRosado** [comment]
When I was updating the file I’ve noticed the fuzzy as well. I believe it happens when other places use a similar log but not exactly the same - because when it’s the same it will show the location of all of them.

I think that’s the reason why the other languages are showing the wrong names for the translation.

Now... do you think we should remove the name of the translated log? I’m asking since the translation will come after the log location like `opsdroid.connector.watson` 🤔

**Comment by jacobtomlinson** [comment]
I don't know enough about how it works to advise. Sorry!

**Comment by FabioRosado** [comment]
So I've been going around the translations and I think I understand what the fuzzy thing is... I think babel/gettext sees some similarities with some logs and will just copy the ones that seem appropriate, but uses fuzzy because the code is not sure if its the right translation.

I'm going through the logs and check them, in some cases I'll try to use the correct names in other cases I'll remove the names altogether which shouldn't be too bad since the logs are being logged with the format `opsdroid.connector.telegram` like I said before.

I'll update this PR soon and merge it. I appreciate your pair of eyes on the PR since you spotted something that I didn't since I focus only on the Portuguese translations haha

**Pull request merged**